### PR TITLE
improve "API" docs

### LINKED ISSUE #644: document across public-facing dependencies
`MixedModels.jl` would be much more user-friendly and accessible to noobs such as myself if the docs included or cross-linked to dependencies that export relevant functions, ideally with a search functionality that spans all such packages.

I keep searching through the docs looking for things that one would expect, but not finding them because they are defined in dependencies of `MixedModels.jl` that I didn't even know exist.

### PR DESCRIPTION
**Pull request by palday**
hopefully fixes #644

- [x] use StatsAPI directly instead of via StatsBase
- [x] move to `using` and explicit overload method definitions instead of `import` 
- [x] have `RegressionModel` and `StatisticalModel` API be obvious in our docs

Did behavior change? Did you add need features? If so, please update NEWS.md
- [ ] add entry in NEWS.md
- [ ] after opening this PR, add a reference and run `docs/NEWS-update.jl` to update the cross-references.

Should we release your changes right away? If so, bump the version:
- [ ] I've bumped the version appropriately

### CODE REVIEW

**Comment by palday** [comment]
@dmbates I've changed from `import` to `using` with explicit annotation of 'owner' modules for functions we're extending with new methods. It makes things a little bit more verbose, but I think it makes it a little bit easier to keep track of when we're doing something new versus extending other definitions. Feel free to object. 😄

**Comment by dmbates** [comment]
Looks good to me.  I am fine with having the qualified names to make it clearer where the generic is defined and documented.  Do you want a review or are you still working on this?

**Comment by palday** [comment]
@dmbates I think I'm ready for review.

File: src/MixedModels.jl [hunk 1]
```diff
@@ -15,23 +15,21 @@ using ProgressMeter
...
 using StatsModels: TableRegressionModel
-
 using StatsFuns: log2π, normccdf
 
-import Base: *
-import DataAPI: levels, refpool, refarray, refvalue
-import GLM: dispersion, dispersion_parameter
-import NLopt: Opt
-import StatsBase: fit, fit!
+import Base: ==
```
Comment (dmbates) [inline src/MixedModels.jl:32]: Why is the import of `==` required?  I couldn't find a case of an `==` method being defined.

Comment (dmbates) [reply src/MixedModels.jl:32]: I think that line is redundant.  I was able to comment it out and still pass the tests.

File: src/linalg/rankUpdate.jl [hunk 2]
```diff
@@ -91,7 +91,7 @@ end
 
 function rankUpdate!(
     C::HermOrSym{T,Diagonal{T,Vector{T}}}, A::StridedMatrix{T}, α, β
-) where {T,S}
+) where {T}
```
Comment (dmbates) [inline src/linalg/rankUpdate.jl:94]: Good catch on this and the following change.

**Review by dmbates — APPROVED** [review]
Just that one change in `MixedModels.jl` and it looks good to go.

File: src/MixedModels.jl (hunk 1 above)
Comment (palday) [reply src/MixedModels.jl:32]: oops, I missed removing it.

Comment (palday) [reply src/MixedModels.jl:32]:
```suggestion
```

File: src/linalg/rankUpdate.jl (hunk 2 above)
Comment (palday) [reply src/linalg/rankUpdate.jl:94]: I received a warning on Julia 1.8.2 when I loaded MixedModels!

**Pull request merged**